### PR TITLE
VIDEO-5531: Support Video Content Preferences

### PR DESCRIPTION
**Pull request by ceaglest**
Adding support for Video Content Preferences. The default (resolves to auto) automatically requests appropriately sized video from the server based upon the size of VideoView. 

Similar to CTSO the manual setting doesn't call the manual control APIs (yet). In this case you will get the server default for bandwidth and resolution based upon track priority.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -106,7 +106,7 @@ PODS:
   - Nimble (9.2.0)
   - PromisesObjC (1.2.12)
   - Quick (3.1.2)
-  - TwilioVideo (4.5.0-rc1)
+  - TwilioVideo (4.5.0-rc4)
```
Comment (ceaglest) [inline Podfile.lock:109]: Needs fixing before merge. I am using the https remote and this branch is setup to use the git remote.

File: VideoApp/Video-InternalTests/Mocks/MockAppSettingsStore.swift [hunk 2]
```diff
@@ -327,72 +328,6 @@ class MockAppSettingsStore: AppSettingsStoreWriting {
         }
     }
 
-    var invokedLowRenderDimensionsSetter = false
```
Comment (ceaglest) [inline VideoApp/Video-InternalTests/Mocks/MockAppSettingsStore.swift:330]: I am not opposed to adding test coverage for the new AppSettingStore properties but I would need some guidance on how to complete this.

File: VideoApp/VideoApp/Stores/AppSettings/VideoContentPreferencesMode.swift [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //
-//  Copyright (C) 2020 Twilio, Inc.
+//  Copyright (C) 2021 Twilio, Inc.
```
Comment (ceaglest) [inline VideoApp/VideoApp/Stores/AppSettings/VideoContentPreferencesMode.swift:2]: Wasn't done as a git rename - github is getting cheeky.

**Review by ceaglest — COMMENTED** [review]
Adding self review of the PR.

File: Podfile.lock (hunk 1 above)
Comment (timrozum) [reply Podfile.lock:109]: Cool shouldn't hold up this PR, just the feature branch.

File: VideoApp/VideoApp/Stores/AppSettings/AppSettingsStore.swift [hunk 4]
```diff
@@ -31,11 +31,9 @@ protocol AppSettingsStoreWriting: LaunchStore {
     var maxSubscriptionBitrate: Int? { get set }
     var dominantSpeakerPriority: TrackPriority { get set }
     var trackSwitchOffMode: TrackSwitchOffMode { get set }
-    var lowRenderDimensions: VideoDimensionsName { get set }
-    var standardRenderDimensions: VideoDimensionsName { get set }
-    var highRenderDimensions: VideoDimensionsName { get set }
```
Comment (timrozum) [inline VideoApp/VideoApp/Stores/AppSettings/AppSettingsStore.swift:36]: This is great.

File: VideoApp/Video-InternalTests/Mocks/MockAppSettingsStore.swift (hunk 2 above)
Comment (timrozum) [reply VideoApp/Video-InternalTests/Mocks/MockAppSettingsStore.swift:330]: I don't think we need to test the settings properties in this app. The main thing is that other parts of the app that we may want to add test coverage for and depend on settings are testable. And that's what this mock is for. 

If you are curious about Swift mocks you can install this Xcode plugin and then use it to regenerate this mock: https://github.com/seanhenry/SwiftMockGeneratorForXcode

It's kind of cool. But if you have any issues just let me know and I can always run the mock generator and push to this branch. Don't want this to take your time.

**Review by timrozum — COMMENTED** [review]
Lets just figure out the mock changes one way or another and this will be good to go.

**Pull request merged**